test/e2e/validation_test.go

```diff
@@ -49,17 +47,40 @@ var (
 			Optional:  false,
 		},
 	}
+	commatrix *types.ComMatrix
 )
 
 const (
 	docCommatrixBaseFilePath = "../../docs/stable/raw/%s.csv"
 	diffFileComments         = "// `+` indicates a port that isn't in the current documented matrix, and has to be added.\n" +
 		"// `-` indicates a port that has to be removed from the documented matrix.\n"
+	commatrixFile      = "communication-matrix.csv"
+	matrixdiffFile     = "matrix-diff-ss"
 	serviceNodePortMin = 30000
 	serviceNodePortMax = 32767
 )
 
 var _ = Describe("Validation", func() {
+	commatrixGenerated := false
+	BeforeEach(func() {
+		if !commatrixGenerated {
```

Comment:
what's the reason not to create it everytime?
also, this can just be
```go
if commatrix != nil {
  return
}
...
```

Reply:
Its 2 tests using same base 
No need to run the command twice